File: apps/java/design-pattern/src/main/java/pattern/observer/Payroll.java
package pattern.observer;

/**
 * 給与計算オブザーバー
 */
public class Payroll implements Observer {

    private String lastNotification;

    @Override
    public void update(Employee employee) {
        lastNotification = employee.getName() + " の給与が " + employee.getSalary() + " に変更されました";
    }

    public String getLastNotification() {
        return lastNotification;
    }
}

File: apps/php/design-pattern/src/Observer/Payroll.php
<?php

declare(strict_types=1);

namespace DesignPattern\Observer;

class Payroll implements Observer
{
    /** @var string[] */
    private array $log = [];

    public function update(Employee $employee): void
    {
        $this->log[] = sprintf(
            '%s の給与を %.0f に変更しました',
            $employee->getName(),
            $employee->getSalary()
        );
    }

    /** @return string[] */
    public function getLog(): array
    {
        return $this->log;
    }
}
